packages/dev/core/src/Rendering/fluidRenderer/fluidRenderer.ts

```diff
@@ -0,0 +1,547 @@
+    }
+
+    /**
+     * Removes a render object from the fluid renderer
+     * @param renderObject the render object to remove
+     * @param removeUnusedTargetRenderer True to remove/dispose of the target renderer if it's not used anymore (default: true)
+     * @returns True if the render object has been found and released, else false
+     */
+    public removeRenderObject(renderObject: IFluidRenderingRenderObject, removeUnusedTargetRenderer = true): boolean {
+        const index = this._renderObjects.indexOf(renderObject);
+        if (index === -1) {
+            return false;
+        }
+
+        renderObject.object.dispose();
+
+        this._renderObjects.splice(index, 1);
+
+        if (removeUnusedTargetRenderer && this._removeUnusedTargetRenderers()) {
+            this._initialize();
+        } else {
+            this._setParticleSizeForRenderTargets();
+        }
+
+        return true;
+    }
+
+    private _sortRenderingObjects(): void {
+        this._renderObjects.sort((a, b) => {
+            return a.object.priority < b.object.priority ? -1 : a.object.priority > b.object.priority ? 1 : 0;
+        });
+    }
+
+    private _removeUnusedTargetRenderers(): boolean {
+        const indexes: { [id: number]: boolean } = {};
+
+        for (let i = 0; i < this._renderObjects.length; ++i) {
+            const targetRenderer = this._renderObjects[i].targetRenderer;
+            indexes[this._targetRenderers.indexOf(targetRenderer)] = true;
+        }
+
+        let removed = false;
+        const newList: Array<FluidRenderingTargetRenderer> = [];
+        for (let i = 0; i < this._targetRenderers.length; ++i) {
+            if (!indexes[i]) {
+                this._targetRenderers[i].dispose();
+                removed = true;
+            } else {
+                newList.push(this._targetRenderers[i]);
+            }
+        }
+
+        if (removed) {
+            this._targetRenderers.length = 0;
+            this._targetRenderers.push(...newList);
+        }
+
+        return removed;
+    }
+
+    private static _IsParticleSystemObject(obj: FluidRenderingObject): obj is FluidRenderingObjectParticleSystem {
+        return !!(obj as FluidRenderingObjectParticleSystem).particleSystem;
+    }
+
+    private _getParticleSystemIndex(ps: IParticleSystem): number {
+        for (let i = 0; i < this._renderObjects.length; ++i) {
+            const obj = this._renderObjects[i].object;
+            if (FluidRenderer._IsParticleSystemObject(obj) && obj.particleSystem === ps) {
+                return i;
+            }
+        }
+
+        return -1;
+    }
+
+    private _initialize(): void {
+        for (let i = 0; i < this._targetRenderers.length; ++i) {
+            this._targetRenderers[i].dispose();
+        }
+
+        const cameras: Map<Camera, [Array<FluidRenderingTargetRenderer>, { [key: string]: FluidRenderingDepthTextureCopy }]> = new Map();
+
+        for (let i = 0; i < this._targetRenderers.length; ++i) {

```

Comment:
That's what I did first (see https://github.com/BabylonJS/Babylon.js/pull/13291/commits/d426583ea8ad9906d0ee9e401d29ff46b58de2d6) but it doesn't work because of our typescript settings. So I reverted to a simple for loop.